.github/workflows/check_unit_tests.yml

```diff
@@ -0,0 +1,42 @@
+name: unit tests check
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.13'
+          cache: 'pip'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install '.[dev]'
+
+      - name: Run unit tests
+        shell: bash
+        run: |
+          pytest --cov=dibbs_text_to_code
+
+      - name: Generate coverage report
+        run: |
+          coverage xml -o coverage.xml
+
+      - name: Upload combined coverage to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
@m-goggins if you log into https://codecov.io with your github account, can you add this repo to codecov?

Reply:
awesome, thank you!  I see it up there now https://app.codecov.io/gh/CDCgov/dibbs-text-to-code/tree/qa%2F6-automations